docs/resources/streamlit.md

```diff
@@ -0,0 +1,106 @@
+---
+page_title: "snowflake_streamlit Resource - terraform-provider-snowflake"
+subcategory: ""
+description: |-
+  Resource used to manage streamlits objects. For more information, check streamlit documentation https://docs.snowflake.com/en/sql-reference/commands-streamlit.
+---
+
+!> **V1 release candidate** This resource was reworked and is a release candidate for the V1. We do not expect significant changes in it before the V1. We will welcome any feedback and adjust the resource if needed. Any errors reported will be resolved with a higher priority. We encourage checking this resource out before the V1 release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0920--v0930) to use it.
+
+# snowflake_streamlit (Resource)
+
+Resource used to manage streamlits objects. For more information, check [streamlit documentation](https://docs.snowflake.com/en/sql-reference/commands-streamlit).
+
+## Example Usage
+
+```terraform
+# basic resource
+resource "snowflake_streamlit" "streamlit" {
+  database  = "database"
+  schema    = "schema"
+  name      = "streamlit"
+  stage     = "streamlit_db.streamlit_schema.streamlit_stage"
+  main_file = "/streamlit_main.py"
```

Comment:
do we always have to start the path with a `/` ? what will happen if this leading slash is not provided? Do we fail? Should we add something in the docs? or should we add it if absent (and suppress a change between with and without)?